src/auto/injector.js

```diff
@@ -337,7 +337,53 @@ function annotate(fn, strictDi, name) {
  *
  * @returns {Array.<string>} The names of the services which the function requires.
  */
-
+/**
+ * @ngdoc method
+ * @name $injector#loadNewModules
+ *
+ * @description
+ *
+ * **This is an dangerous API, which you use at your own risk!**
+ *
+ * Add the specified modules to the current injector.
+ *
+ * This method will add each of the injectables to the injector and execute all of the run blocks
+ * for each module passed to the method.
+ * * The application developer is responsible for loading the code containing the modules.
+ * * Modules cannot be unloaded.
+ * * Previously compiled HTML will not be affected by newly loaded directives, filters and components.
+ *
+ * ## Example
+ *
+ * Here is an example of loading a bundle of modules, with a utility method called `getScript`:
+ *
+ * ```javascript
+ * app.service('AmdLoader', function ($injector, $log) {
+ *
+ *  return function(name, config) {
+ *    getScript(config).done(function () {
+ *      $log.info('Script loaded : ' + name);
+ *      onReady(name);
+ *    });
+ *
+ *    function onReady(name) {
+ *      var moduleName = name + '.html.bundle';
+ *      $log.debug('AMD >> Loading Angular module [' + moduleName + '] ...');
+ *
+ *      if (config.modules) {
+ *        $injector.loadNewModules(config.modules);
```

Comment:
Why this? Shouldn't the lazy-loaded module depend on these (as in `angular.module('someModule', [...dependent modules here...])`), in which case `loadNewModules` would automatically load them?

Reply:
Not sure to be honest, this was a bit of a cut-n-paste job.